diff -ur ./build/orig_470.256.02/kernel/common/inc/nv-linux.h ./build/NVIDIA-Linux-x86_64-470.256.02/kernel/common/inc/nv-linux.h
--- a/kernel/common/inc/nv-linux.h
+++ b/kernel/common/inc/nv-linux.h
@@ -1817,7 +1817,7 @@
 #if defined(NV_FILE_HAS_INODE)
 #define NV_FILE_INODE(file) (file)->f_inode
 #else
-#define NV_FILE_INODE(file) (file)->f_dentry->d_inode
+#define NV_FILE_INODE(file) file_inode(file)
 #endif
 
 #if defined(NV_DOM0_KERNEL_PRESENT) || defined(NV_VGPU_KVM_BUILD)
diff -ur ./build/orig_470.256.02/kernel/common/inc/nv-mm.h ./build/NVIDIA-Linux-x86_64-470.256.02/kernel/common/inc/nv-mm.h
--- a/kernel/common/inc/nv-mm.h
+++ b/kernel/common/inc/nv-mm.h
@@ -187,7 +187,7 @@
 #if defined(NV_VM_FAULT_HAS_ADDRESS)
     return vmf->address;
 #else
-    return (unsigned long)(vmf->virtual_address);
+    return (unsigned long)(vmf->address);
 #endif
 }
 
@@ -258,12 +258,12 @@
 #else
 static inline void nv_vm_flags_set(struct vm_area_struct *vma, unsigned long flags)
 {
-    vma->vm_flags |= flags;
+    vm_flags_set(vma, flags);
 }
 
 static inline void nv_vm_flags_clear(struct vm_area_struct *vma, unsigned long flags)
 {
-    vma->vm_flags &= ~flags;
+    vm_flags_clear(vma, flags);
 }
 #endif
 
diff -ur ./build/orig_470.256.02/kernel/common/inc/nv-time.h ./build/NVIDIA-Linux-x86_64-470.256.02/kernel/common/inc/nv-time.h
--- a/kernel/common/inc/nv-time.h
+++ b/kernel/common/inc/nv-time.h
@@ -141,7 +141,7 @@
     ktime_get_real_ts64(&tm1);
 #endif
 
-    if (in_irq() && (us > NV_MAX_ISR_DELAY_US))
+    if (in_hardirq() && (us > NV_MAX_ISR_DELAY_US))
         return NV_ERR_GENERIC;
 
     mdelay_safe_msec = us / 1000;
@@ -187,7 +187,7 @@
     tm_start = tm_aux;
 #endif
 
-    if (in_irq() && (ms > NV_MAX_ISR_DELAY_MS))
+    if (in_hardirq() && (ms > NV_MAX_ISR_DELAY_MS))
     {
         return NV_ERR_GENERIC;
     }
